pkg/controller/proposal/controller.go

```diff
@@ -421,6 +399,55 @@ func (r *Reconciler) reconcileValidate(ctx context.Context, proposal *configapi.
 	}
 }
 
+func (r *Reconciler) reconcileAbort(ctx context.Context, proposal *configapi.Proposal) (controller.Result, error) {
+	switch proposal.Status.Phases.Abort.State {
+	case configapi.ProposalAbortPhase_ABORTING:
+		configID := configuration.NewID(proposal.TargetID)
+		config, err := r.configurations.Get(ctx, configID)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				log.Errorf("Failed reconciling Transaction %d Proposal to target '%s'", proposal.TransactionIndex, proposal.TargetID, err)
+				return controller.Result{}, err
+			}
+			return controller.Result{}, nil
+		}
+		if config.Status.Committed.Index == proposal.Status.PrevIndex {
+			config.Status.Committed.Index = proposal.TransactionIndex
+			if err := r.configurations.UpdateStatus(ctx, config); err != nil {
+				log.Errorf("Failed reconciling Transaction %d Proposal to target '%s'", proposal.TransactionIndex, proposal.TargetID, err)
+				return controller.Result{}, err
+			}
+		} else if config.Status.Applied.Index == proposal.Status.PrevIndex &&
+			config.Status.Committed.Index >= proposal.TransactionIndex {
+			config.Status.Committed.Index = proposal.TransactionIndex
+			if err := r.configurations.UpdateStatus(ctx, config); err != nil {
+				log.Errorf("Failed reconciling Transaction %d Proposal to target '%s'", proposal.TransactionIndex, proposal.TargetID, err)
+				return controller.Result{}, err
+			}
+			proposal.Status.Phases.Abort.End = getCurrentTimestamp()
+			proposal.Status.Phases.Abort.State = configapi.ProposalAbortPhase_ABORTED
+			if err := r.updateProposalStatus(ctx, proposal); err != nil {
+				return controller.Result{}, err
+			}
+			return controller.Result{}, nil
+		} else if config.Status.Committed.Index == proposal.Status.PrevIndex &&
```

Comment:
This should be the first condition, otherwise we're always going to (unnecessarily) update the configuration twice.